Installer/MainWindow.xaml.cs

```diff
@@ -0,0 +1,257 @@
+            shortcut.Save();
+        }
+
+        private void SwitchPage()
+        {
+            aboutPage.Visibility = Visibility.Hidden;
+            folderPage.Visibility = Visibility.Hidden;
+            confirmPage.Visibility = Visibility.Hidden;
+            extractingPage.Visibility = Visibility.Hidden;
+            finishedPage.Visibility = Visibility.Hidden;
+            switch (currentPage)
+            {
+                case Page.About:
+                    aboutPage.Visibility = Visibility.Visible;
+                    break;
+                case Page.SelectFolder:
+                    if (string.IsNullOrEmpty(vtFolder))
+                        vtFolder = FindVTOL();
+                    folderBox.Text = vtFolder;
+                    folderPage.Visibility = Visibility.Visible;
+                    break;
+                case Page.Confirm:
+                    confirmPage.Visibility = Visibility.Visible;
+                    break;
+                case Page.Extracting:
+                    extractingPage.Visibility = Visibility.Visible;
+                    InstallFiles();
+                    break;
+                case Page.Finished:
+                    finishedPage.Visibility = Visibility.Visible;
+                    cancelButton.Visibility = Visibility.Hidden;
+                    backButton.Content = "Launch";
+                    nextButotn.Content = "Close";
+                    break;
+            }
+        }
+
+        private void CancelButton_Click(object sender, RoutedEventArgs e)
+        {
+            Quit();
+        }
+        private void Quit()
+        {
+            Process.GetCurrentProcess().Kill();
+        }
+
+        private void NextButotn_Click(object sender, RoutedEventArgs e)
+        {
+            if (currentPage == Page.Finished)
+                Quit();
+            if (currentPage == Page.Extracting)
+                return;
+            currentPage++;
+            SwitchPage();
+        }
+
+        private void BackButton_Click(object sender, RoutedEventArgs e)
+        {
+            if (currentPage <= 0)
+                return;
+            if (currentPage == Page.Finished)
+            {
+                ProcessStartInfo startInfo = new ProcessStartInfo(vtFolder + @"VTOLVR_ModLoader\VTOLVR-ModLoader.exe");
+                startInfo.WorkingDirectory = vtFolder + @"VTOLVR_ModLoader\";
+                Process.Start(startInfo);
+                Quit();
+            }
+            currentPage--;
+            SwitchPage();
+        }
+
+        private void BrowseButton_Click(object sender, RoutedEventArgs e)
+        {
+            OpenFileBrowser();
+        }
+        private void SetProgress(float barValue)
+        {
+            progressBar.Value = barValue;
+        }
+
+        private void OpenFileBrowser()
+        {
+            OpenFileDialog openFileDialog = new OpenFileDialog();
+            openFileDialog.InitialDirectory = "C:\\";
```

Comment:
What if the user doesn't have a C drive? 🤔

Reply:
I'm not sure how it handles it but I think it could cause a crash

Reply:
That's a good point. After a test, it seems to just crash the program if it can't find that path.
Changed it to `openFileDialog.InitialDirectory = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles);`